server/src/main/java/org/elasticsearch/index/mapper/PassThroughObjectMapper.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.Explicit;
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.index.IndexVersion;
+import org.elasticsearch.index.mapper.MapperService.MergeReason;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.elasticsearch.common.xcontent.support.XContentMapValues.nodeBooleanValue;
+
+/**
+ * Mapper for pass-through objects.
+ *
+ * Pass-through objects allow creating fields inside them that can also be referenced directly in search queries.
+ * They also include parameters that affect how nested fields get configured. For instance, if parameter "time_series_dimension"
+ * is set, all keyword subfields are marked as dimensions and included in routing and tsid calculations.
+ */
+public class PassThroughObjectMapper extends ObjectMapper {
+    private static final DeprecationLogger DEPRECATION_LOGGER = DeprecationLogger.getLogger(PassThroughObjectMapper.class);
+
+    public static final String CONTENT_TYPE = "passthrough";
+
+    public static class Builder extends ObjectMapper.Builder {
+
+        // Controls whether subfields are configured as time-series dimensions.
+        protected Explicit<Boolean> containsDimensions = Explicit.IMPLICIT_FALSE;
+
+        public Builder(String name) {
+            // Subobjects are not currently supported.
+            super(name, Explicit.IMPLICIT_FALSE);
```

Comment:
While this sets the default to false, I don't think this prevents someone from explicitly setting `subobjects` to `true`.

Reply:
True. The same applies to the PassthroughObjectMapper constructor, making sure that the constructed object is initialized with the corresponding param set to false.

The value here is not needed as it's not used anywhere, but set for consistency.

Reply:
I see. Could you add a yaml rest test to check that explicitly setting `subobjects: false` results in an error?

Reply:
Done.